app/services/question_formatter/blackboard_service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module QuestionFormatter
+  class BlackboardService < BaseService
+    # Even though this is really a TSV under the hood, Blackboard exports out a .txt extension
+    self.output_format = 'txt'
+
+    def format_content
+      case @question
+      when Question::Traditional, Question::SelectAllThatApply
+        traditional_type
```

Comment:
You should use the class's model_exporter, not a case structure. Method format_by_type uses the classes to look up and do this automatically. 

Since you are apparently only handling certain question types, maybe there should be a new class method to define which export types are supported? I'd suggest that you add the values you use in type_mapper as "blackboard_export_type" and nil for the classes which we don't export to blackboard. Then this method becomes:

```
def format_content
blackboard_type = @question.class.blackboard_export_type
if !blackboard_type.nil?
  format_by_type
  [@question.class.model_exporter, @text, @answers].join('\t')
end
```